casper/src/main/rholang/WalletCheck.rho

```diff
@@ -1,20 +1,23 @@
+//scalapackage coop.rchain.rholang.wallet
+
 // WalletCheck: holds funds until a public key with the given hash is presented,
 //     then creates a wallet with that public key and those funds.
 
 // create: makes a wallet check
 //
-// bytearray hash: the hash of the public key
+// string ethAddr: the address of the funds on Ethereum (pre-issuance)
 // Purse purse: the funds to put in the wallet
-contract @["WalletCheck", "create"](@hash, @purse) = {
+contract @["WalletCheck", "create"](@ethAddr, @purse) = {
   new doneStore in {
     doneStore!(false) |
     
     // claims the wallet by presenting the preimage of the hash
     //
-    // bytearray pubkey
+    // string pubkey: Base16 encoded public key. It should be true that
+    //                "0x" + Base16.encode(keccak256.hash(Base16.decode(pubkey)).takeRight(20)) == ethAddr
     // name statusOut
-    // bytearray sig: signature of [pubKey, statusOut]
-    contract @hash(@[pubKey, statusOut], @sig) = {
+    // string sig: signature of [pubKey, statusOut]
+    contract @ethAddr(@[pubKey, statusOut], @sig) = {
```

Comment:
Question about rholang style. Why you don't use tuples for small sets of values?

Reply:
It's a good point, we really should be using tuples. It's more of a habit for me because tuples didn't exist when I started writing Rholang.